curvefs/src/client/s3/disk_cache_manager.cpp

```diff
@@ -359,28 +363,15 @@ void DiskCacheManager::TrimCache() {
                 if (ret == 0) {
                     VLOG(3) << "do not remove this disk file"
                             << ", file has not been uploaded to S3."
-                            << ", file is: " << *cacheKeyIter;
-                    cachedObjNameTmp.erase(cacheKeyIter);
-                    continue;
-                }
-                {
-                    std::lock_guard<bthread::Mutex> lk(mtx_);
-                    std::set<std::string>::iterator iter;
-                    iter = cachedObjName_.find(*cacheKeyIter);
-                    if (iter == cachedObjName_.end()) {
-                        VLOG(6) << "remove disk file error"
-                                << ", file is not exist in cachedObjName"
-                                << ", file is: " << *cacheKeyIter;
-                        cachedObjNameTmp.erase(cacheKeyIter);
-                        continue;
-                    }
-                    cachedObjName_.erase(iter);
+                            << ", file is: " << cacheKey;
+                    break;
                 }
+                cachedObjName_->Remove(cacheKey);
```

Comment:
if we remove cachekey first, then come a read, will it cause an concurrency problem?